<launch>
    <!-- Face detector node -->
    <node pkg="turtlebot3_face_recognition" 
          type="detect_face_image.py" 
          name="image_face_detector" 
          output="screen">
        <param name="haar_path" value="$(find turtlebot3_face_recognition)/data/haarcascade_frontalface_default.xml"/>
        <param name="image_folder" value="$(find turtlebot3_face_recognition)/data/"/>
    </node>

    <!-- Image viewer with full environment configuration -->
    <node pkg="image_view" 
          type="image_view" 
          name="image_view"
          output="screen"
          launch-prefix="bash -c 'export DISPLAY=${DISPLAY:-:0}; export QT_X11_NO_MITSHM=1; export XAUTHORITY=~/.Xauthority; $0 $@' ">
        <remap from="image" to="/image_face_detector/output_image"/>
    </node>
</launch>

